aip/0009.md

```diff
@@ -0,0 +1,99 @@
+---
+aip:
+  id: 9
+  state: draft
+  created: 2019-08-01
+  modified: 2019-08-04
+permalink: /9
+redirect_from:
+  - /09
+  - /009
+  - /0009
+---
+
+# AIP Glossary
+
+In the name of brevity, this AIP defines some common terminology here rather
+than in each AIP individually.
+
+## Guidance
+
+The following terminology **should** be used consistently throughout AIPs.
+
+API
+: Application Programming Interface. This can be a local interface (such as a
+  client library) or a Network API (defined below).
+
+Network API
+: An API that operate across a network of computers. Network APIs communicate
+  using network protocols including HTTP, and are frequently produced by
+  organizations separate from those that consume them.
+
+Google API
+: A Network API exposed by a Google service. Most of these are hosted on the
+  googleapis.com domain. It does not include other types of APIs, such as client
+  libraries and SDKs.
+
+API Definition
+: The defintion of an API, usually defined in a Protocol Buffer service. It is
+  typically mapped to an interface in most programming languages. An API
+  Definition can be implemented by any number of API Services.
+
+API Version
+: The version of an API or a group of APIs if they are defined together. An API
+  Version is often represented by a string, such as "v1", and presents in API
+  requests and Protocol Buffers package names.
+
+API Method
+: An individual operation within an API. It is typically represented in Protocol
+  Buffers by an `rpc` definition, and is mapped to a function in the API in most programming languages.
+
+API Request
+: A single invocation of an API Method. It is often used as the unit for
+  billing, logging, monitoring, and rate limiting.
+
+API Consumer
+: The entity that consumes an API Service. For Google APIs, it typically is a
+  Google project that owns the client application or the server resource.
+
+API Producer
+: The entity that produces an API Service. For Google APIs, it typically is a
+  Google team responsible for the API Service.
+
+API Backend
+: A set of servers and related infrastructure that implements the business
+  logic for an API Service. An individual API backend server is often called an
+  API server.
+
+API Frontend
+: A set of servers plus related infrastructure that provides common
+  functionality across API Services, such as load balancing and authentication.
+  An individual API frontend server is often called an API proxy.
+  NOTE: the API frontend and the API backend may run next to each other or far
+  away from each other. In some cases, they can be compiled into a single
+  application binary and run inside a single process.
+
+API Service
+: A deployed implementation of one or more APIs, exposed on one or more network
+  addresses, such as the Cloud Pub/Sub API.
+
+API Service Name
```

Comment:
We should clarify the plurality here. For example, is PubSub a single API service? What do we call "Publisher" and "Subscriber"?